packages/ui/src/coreModules/form/higherOrderComponents/createLinkFieldErrors.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { omit } from 'lodash'
+import {
+  clearSubmitErrors as clearSubmitErrorsActionCreator,
+  formValueSelector,
+  getFormSubmitErrors,
+} from 'redux-form'
+import { connect } from 'react-redux'
+
+const propTypes = {
+  clearSubmitErrors: PropTypes.func.isRequired,
+  formName: PropTypes.string.isRequired,
+  linkedFieldErrorValues: PropTypes.object,
+}
+
+const defaultProps = {
+  linkedFieldErrorValues: undefined,
+}
+
+const mapDispatchToProps = {
+  clearSubmitErrors: clearSubmitErrorsActionCreator,
+}
+// Note this is a first version of this component.
+// Atm it only supports fieldPaths without dots (otherwise object-path has to be used)
+// To work with multiple groups of fields the component has to be applied multiple times
+
+export default function createLinkFieldErrors({ fieldPaths = [] } = {}) {
+  const mapStateToProps = (state, { form: formName }) => {
+    const linkedFieldErrorValues = {}
+    const submitErrors = getFormSubmitErrors(formName)(state)
+    let anyLinkedFieldHasErrors = false
+    fieldPaths.forEach(fieldPath => {
+      const fieldValue = formValueSelector(formName)(state, fieldPath)
+      if (submitErrors[fieldPath]) {
+        anyLinkedFieldHasErrors = true
+      }
+      linkedFieldErrorValues[fieldPath] = fieldValue
+    })
+    if (!anyLinkedFieldHasErrors) {
```

Comment:
I generally think newlines between segments of code (e.g. before the forEach and before the if) makes it easier to read, but that might just be a personal preference

Reply:
sure